cmd/maya-apiserver/cstor-operator/cspc/runner.go

```diff
@@ -95,26 +104,43 @@ func (c *Controller) processNextWorkItem() bool {
 			// process a work item that is invalid.
 			c.workqueue.Forget(obj)
 			runtime.HandleError(fmt.Errorf("expected string in workqueue but got %#v", obj))
-			return nil
+			return nil, ""
 		}
 		// Run the syncHandler, passing it the namespace/name string of the
 		// Foo resource to be synced.
 		if err := c.syncHandler(key); err != nil {
 			// Put the item back on the workqueue to handle any transient errors.
 			c.workqueue.AddRateLimited(key)
-			return fmt.Errorf("error syncing '%s': %s, requeuing", key, err.Error())
+			return fmt.Errorf("error syncing '%s': %s", key, err.Error()), key
 		}
 		// Finally, if no error occurs we Forget this item so it does not
 		// get queued again until another change happens.
 		c.workqueue.Forget(obj)
 		glog.V(1).Infof("Successfully synced '%s'", key)
-		return nil
+		return nil, key
 	}(obj)
 
 	if err != nil {
-		runtime.HandleError(err)
+		c.handleErr(err, key)
```

Comment:
Let's avoid multiple checks for `err`. `handleErr` is already checking if `err` is `nil` or not.

Reply:
done